.github/workflows/ci.yml

```diff
@@ -0,0 +1,23 @@
+name: build ci
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Set up Go
+      uses: actions/setup-go@v3
+      with:
+        go-version: 1.19
+
+    - name: build
+      run: make build
```

Comment:
need run tests.

Reply:
Also, need a better way to test other modes, now we could only test with simple static configuration.
we should test with different configuration modes, eventually, not a very high priority.

Reply:
I will fix this later.